### ISSUE TITLE
Unexpected, though Unsurprising, Rendering of XML Comment Embedded in Triple-Slash Comment

### ISSUE THREAD

**Issue by txwizard** [dotnet/docfx#3493]
**DocFX Version Used**: 
2.39.1.0
**Template used**: (`default` or `statictoc` or contain custom template)
default
**Steps to Reproduce**:

1.  Create an Example tag in the triple-slash documentation that resembles the following.

 		/// <example>
		/// Parse this: 
		/// 
		/// Return this:
		/// 
		///			=======================
		///			Name		Value
		///			-----------	-----------
		///			ForPage		default
		///			UseTable	False
		///			=======================
		///			
		/// The returned NameValueCollection contains two members.
		/// 
		/// Since this method guarantees to return an initialized
		/// NameValueCollection, the empty collection is allocated by the first
		/// statement, and is unconditionally returned by the last statement.
		/// </example>

2.  Generate metadata as usual, 
`docfx.exe $(SolutionDir)docfx_project\docfx.json --metadata`

**Expected Behavior**:

> rse this: 

Return this:

		///			=======================
		///			Name		Value
		///			-----------	-----------
		///			ForPage		default
		///			UseTable	False
		///			=======================
		///
The returned NameValueCollection contains two members.

Since this method guarantees to return an initialized
NameValueCollection, the empty collection is allocated by the first
statement, and is unconditionally returned by the last statement.

**Actual Behavior**:

>   <h5 id="WizardWrx_StringExtensions_ParseCommentInHTMLComment_System_String__examples">Examples</h5>
  <p>Parse this: </p>
<p>Return this:</p>
<p>=======================
Name		Value</p>
<hr>
<h1 id="forpagedefaultusetablefalse">ForPage		default
UseTable	False</h1>
<p>The returned NameValueCollection contains two members.</p>
  <h5 id="WizardWrx_StringExtensions_ParseCommentInHTMLComment_System_String__examples">Examples</h5>
  <p>Parse this: </p>
<p>Return this:</p>
<p>=======================
Name		Value</p>
<hr>
<h1 id="forpagedefaultusetablefalse">ForPage		default
UseTable	False</h1>
<p>The returned NameValueCollection contains two members.</p>
<p>Since this method guarantees to return an initialized
NameValueCollection, the empty collection is allocated by the first
statement, and is unconditionally returned by the last statement.</p>

The entire example is in the documentation that I generated for the WizardWrx .NET API and published at [https://txwizard.github.io/WizardWrx_NET_API/api/WizardWrx.StringExtensions.html](url), and the repository is at [https://github.com/txwizard/WizardWrx_NET_API](url), where you can find the DocFX metadata in `/docfx_project`, and the generated HTML in `/docs`, so that I can point the GitHub Pages engine at it.

The build log gives no indication that the XML comment raised any alarms.

> F:\Source_Code\Visual_Studio\Projects\WizardWrx_Libs\Wizardwrx 2018/09/05  1:23:08.25>docfx docfx_project\docfx.json --metadata
[18-09-05 06:23:13.678]Info:[MetadataCommand.ExtractMetadata]Using msbuild C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin as inner compiler.
[18-09-05 06:23:13.928]Info:[MetadataCommand.ExtractMetadata]Loading projects...
[18-09-05 06:23:16.959]Info:[MetadataCommand.ExtractMetadata]Generating metadata for each project...
[18-09-05 06:23:18.552]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/ASCIIInfo/ASCIIInfo.csproj' keep up-to-date since '2018/09/05 05:41:50', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/ASCIIInfo/obj/xdoc/cache/obj\2ufbefq3.tp5' is used.
[18-09-05 06:23:18.677]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/Common/Common.csproj' keep up-to-date since '2018/09/05 05:41:49', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/Common/obj/xdoc/cache/obj\gemohdju.e53' is used.
[18-09-05 06:23:18.864]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/FormatStringEngine/FormatStringEngine.csproj' keep up-to-date since '2018/09/05 05:41:50', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/FormatStringEngine/obj/xdoc/cache/obj\hy0tkvll.5is' is used.
[18-09-05 06:23:18.958]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/Core/Core.csproj' keep up-to-date since '2018/09/05 05:41:50', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/Core/obj/xdoc/cache/obj\eoa25yg1.1b5' is used.
[18-09-05 06:23:19.286]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/AssemblyUtils/AssemblyUtils.csproj' keep up-to-date since '2018/09/05 05:41:51', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/AssemblyUtils/obj/xdoc/cache/obj\nertdkcb.zlu' is used.
[18-09-05 06:23:19.333]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/ConsoleStreams/ConsoleStreams.csproj' keep up-to-date since '2018/09/05 05:41:51', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/ConsoleStreams/obj/xdoc/cache/obj\2irvipgt.e1b' is used.
[18-09-05 06:23:19.458]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/DLLConfigurationManager/DLLConfigurationManager.csproj' keep up-to-date since '2018/09/05 05:41:51', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/DLLConfigurationManager/obj/xdoc/cache/obj\134jlxmw.gy1' is used.
[18-09-05 06:23:19.552]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/EmbeddedTextFile/EmbeddedTextFile.csproj' keep up-to-date since '2018/09/05 05:41:52', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/EmbeddedTextFile/obj/xdoc/cache/obj\mice0t0h.kst' is used.
[18-09-05 06:23:19.567]Info:[MetadataCommand.ExtractMetadata]'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/DLLServices2TestStand/DLLServices2TestStand.csproj' keep up-to-date since '2018/09/05 05:41:52', cached intermediate result 'F:/Source_Code/Visual_Studio/Projects/WizardWrx_Libs/Wizardwrx/DLLServices2TestStand/obj/xdoc/cache/obj\opyanfks.wfd' is used.
[18-09-05 06:23:21.708]Info:[MetadataCommand]Completed Scope:MetadataCommand in 8054.2975 milliseconds.
[18-09-05 06:23:21.817]Info:[BuildCommand]6 plug-in(s) loaded.
[18-09-05 06:23:21.848]Info:[BuildCommand]No files are found with glob pattern apidoc/**.md, excluding obj/**,_site/**, under directory "F:\Source_Code\Visual_Studio\Projects\WizardWrx_Libs\Wizardwrx\docfx_project"
[18-09-05 06:23:21.848]Info:[BuildCommand]No files are found with glob pattern images/**, excluding <none>, under directory "F:\Source_Code\Visual_Studio\Projects\WizardWrx_Libs\Wizardwrx\docfx_project"
[18-09-05 06:23:21.864]Info:[BuildCommand]Markdown engine is markdig
[18-09-05 06:23:22.145]Info:[BuildCommand.BuildCore.Build Document]Max parallelism is 8.
[18-09-05 06:23:22.176]Info:[BuildCommand.BuildCore.Build Document.Prepare.CreateIncrementalBuildContext]Build strategy: IsFullBuild
[18-09-05 06:23:24.488]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ManagedReferenceDocumentProcessor]Building 104 file(s) in ManagedReferenceDocumentProcessor(BuildManagedReferenceDocument=>ValidateManagedReferenceDocumentMetadata=>ApplyOverwriteDocumentForMref=>FillReferenceInformation)...
[18-09-05 06:23:24.488]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ConceptualDocumentProcessor]Building 3 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
[18-09-05 06:23:24.504]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.TocDocumentProcessor]Building 3 file(s) in TocDocumentProcessor(BuildTocDocument)...
[18-09-05 06:23:46.577]Info:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.UpdateContext]0 external references found in 1 xref maps.
[18-09-05 06:23:48.061]Info:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.Apply Templates]Applying templates to 110 model(s)...
[18-09-05 06:23:50.701]Info:[BuildCommand.BuildCore.Build Document]XRef map exported.
[18-09-05 06:23:52.403]Info:[BuildCommand.Postprocess]Manifest file saved to manifest.json.
[18-09-05 06:23:52.513]Info:[BuildCommand]Completed building documents in 30689.861 milliseconds.
[18-09-05 06:23:52.513]Info:[BuildCommand]Completed Scope:BuildCommand in 30806.4418 milliseconds.
[18-09-05 06:23:52.528]Info:Completed in 38869.348 milliseconds

I would have expected the generator to raise an alarm.

**Comment by superyyrrzz** [comment]
The content in `<example>` is taken as a piece of Markdown content, so your text is unintentionally transformed, e.g. `<!-- ForPage=default;UseTable=False -->` is missing as as its HTML comment, `ForPage	default UseTable	False` becomes a title with the belown `========`. 

One solution is to try wrap your text that unintentionally matches Markdown syntax with code block, like
```
/// <example>
/// Parse this: `<!-- ForPage=default;UseTable=False -->`
/// 
/// Return this:
/// ```
///			=======================
///			Name		Value
///			-----------	-----------
///			ForPage		default
///			UseTable	False
///			=======================
/// ```		
/// ......
/// </example>
```

**Comment by herohua** [comment]
@txwizard have you tried the solution provided by @superyyrrzz ?

**Comment by rjgotten** [comment]
> The content in `<example>` is taken as a piece of Markdown content

And why is DocFX doing that?
Using Markdown in conceptual articles, written specifically for a DocFX project; sure.
But inside API reference distilled from .NET XML Documentation?

That sounds like all kinds of wrong and just has "bugs waiting to happen" written all over it.

**Comment by superyyrrzz** [comment]
@rjgotten DocFX provides `--shouldSkipMarkup` option to disable this feature. It will keep XML doc in its original text. #756

**Comment by txwizard** [comment]
@superyyrrzz, please accept my apology for not replying sooner to your excellent suggestion. Since I've spent all of this week working on other projects that, coincidentally, use libraries from this project, I haven't had even a spare minute to log into GitHub, for any reason, until today.

Thankfully, I discovered the **code** tag in the course of researching an unrelated matter, and succeeded in completely resolving the formatting issue that prompted me opening this issue. Wrapping code tags around the offending text, along with substituting XML entities for the embedded angle brackets, yielded the expected outcome shown below.

![embedded_xml_comment_fixed_20181117_162605](https://user-images.githubusercontent.com/19792705/48666373-cb782a00-ea85-11e8-8984-850ed411f180.PNG)
Assuming that it becomes visible in the finished product, the above image shows the amended page that will shortly make its way into the repository.

Before I wrap up, I respectfully disagree with the suggestion by @rjgotten that the example has "all kinds of wrong" with it. The example is essential to communicating how to use the associated method; that's my story, and I'm sticking to it.

**Comment by rjgotten** [comment]
> @txwizard 
>  I respectfully disagree with the suggestion by @rjgotten that the example has "all kinds of wrong" with it. The example is essential to communicating how to use the associated method

If you're writing XMLDoc and you need to put a table into your example, you can do that with native XMLDoc. Using `<list type="table">` tags. That's what they're there for. (Well... if they would work as according to the official published documentation. Which they don't quite.)

The point I'm making is that DocFX should not _in general_ be parsing text content inside XMLDoc tags as Markdown. _IF_ you want to opt-in to that; introduce a `<markdown>` tag or something.

**Comment by txwizard** [comment]
Thank you, @rjgotten, for calling attention to the native mechanism in XMLDoc for creating tables. I discovered that capability some time after I closed this ticket,  Having now seen your clarification, for which I also owe you my seriously belated thanks, I am happy to report as well that I eventually found my way to complete documentation of XMLDoc, where I discovered the options for including tables, lists, and all sorts of other goodies.

I agree with you that DocFX has no business parsing text inside XMLDoc tags. The origin of my initial surprise was that it seemed to be attempting to do so, with very strange consequences. I expected it to be treated as if it were wrapped in &lt;pre:&gt; tags.
